python/TestHarness/validation/ValidationCase.py

```diff
@@ -0,0 +1,343 @@
+        self._current_not_validation: Optional[bool] = None
+        # The current outputs from the test run, if any
+        self._tester_outputs: Optional[list] = tester_outputs
+
+    def addResult(self, status: Status, message: str, **kwargs) -> None:
+        """
+        Adds a validation result to the database.
+
+        Args:
+            status: The status (ok, fail, etc)
+            message: Message associated with the status
+        Keyword arguments:
+            Additional arguments passed to a Result
+        """
+        assert isinstance(status, self.Status)
+        assert isinstance(message, str)
+        status_value = self.Result(status=status, message=message,
+                                   test=self._current_test, **kwargs)
+
+        prefix = ''
+        if status_value.data_key is not None:
+            data = self._data[status_value.data_key]
+            prefix = f'{data.key} = {data.printableValue()} '
+        print(f'[{status.value:>4}] {prefix}{message}')
+        self._results.append(status_value)
+
+    def addFloatData(self, key: str, value: float, units: Optional[str],
+                     description: str, **kwargs):
+        """
+        Adds a piece of float data to the validation data.
+
+        Will also perform checking on the data if bounds are set and
+        store an associated Result.
+
+        Args:
+            key: The key to store the data
+            value: The value of the data
+            units: Human readable units for the data (can be None)
+            description: Human readable description of the data
+        Keyword arguments:
+            Additional arguments passed to FloatData
+        """
+        if not isinstance(value, float):
+            raise ValueError('value is not of type float')
+
+        if key in self._data:
+            raise DataKeyAlreadyExists(key)
+
+        data = self.FloatData(value=value,
+                              units=units,
+                              key=key,
+                              description=description,
+                              test=self._current_test,
+                              **kwargs)
+        self._data[key] = data
+
+        result_kwargs = {'data_key': key,
+                         'validation': kwargs.pop('validation', True)}
+        units = f' {data.units}' if data.units is not None else ''
+
+        if data.bounds is not None:
+            assert len(data.bounds) == 2
+            min, max = data.bounds
+            assert max > min
+            success = value >= min and value <= max
+            status = self.Status.OK if success else self.Status.FAIL
+            message = [('within' if success else 'out of') + ' bounds;']
+            message += [f'min = {min:.5E}{units},']
+            message += [f'max = {max:.5E}{units}']
+            self.addResult(status, ' '.join(message), **result_kwargs)
+
+    @property
+    def results(self) -> list['Result']:
```

Comment:
Is this what you meant?
```suggestion
    def results(self) -> list[Result]:
```